src/java.desktop/macosx/classes/sun/lwawt/macosx/CPlatformResponder.java

```diff
@@ -79,6 +79,14 @@ void handleMouseEvent(int eventType, int modifierFlags, int buttonNumber,
         }
 
         int jmodifiers = NSEvent.nsToJavaModifiers(modifierFlags);
+        if ((jeventType == MouseEvent.MOUSE_PRESSED) && (jbuttonNumber > MouseEvent.NOBUTTON)) {
+            // 8294426: NSEvent.nsToJavaModifiers returns 0 on M2 MacBooks if the event is generated
```

Comment:
You say it returns 0, so you expect to only need to do that if jmodifiers is 0, yet you don't check that.
Why not ?

No way to test this on an M2 so being sure it doesn't regress something is all I can offer here.

Reply:
Actually I mean that it returns 0 if you're tapping without holding any modifiers e.g. keyboard modifiers.
But if you're tapping holding a keyboard modifier it will probably return only this keyboard modifier without `InputEvent.BUTTON3_DOWN_MASK`. I didn't test this case but I would expect it that looking at the implementation of the `NSEvent.nsToJavaModifiers`.

Reply:
I don't have an M2 device to check, but I think (judging to the code) if we add `jmodifiers == 0` check here then users making a tap holding the Control will receive the mouse events with `modifiers` == `CTRL_DOWN_MASK | BUTTON3_MASK | CTRL_MASK`, i.e. without modern `BUTTON3_DOWN_MASK` which is at least unusual and inconvenient.

Reply:
OK .. well I've at least run all our automated tests and no regressions were found, so I'll approve this.

Reply:
I've checked behavior of tap holding modifiers on M2 and here are which mouse events are received:
* _Ctrl + Two fingers tap_:
  * event: `java.awt.event.MouseEvent[MOUSE_PRESSED,(86,102),absolute(86,302),button=3,modifiers=⌘+⌃+Button3,extModifiers=⌃,clickCount=1] on MouseWindow$1[,0,136,200x200,layout=java.awt.FlowLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777225,maximumSize=,minimumSize=,preferredSize=]`
  * modifiers: 6 (`⌘+⌃+Button3`)
  * modifiersEx: 128 (`⌃`)
* _Option + Two fingers tap_:
  * event: `java.awt.event.MouseEvent[MOUSE_PRESSED,(63,135),absolute(63,335),button=3,modifiers=⌥+⌘+Button2+Button3,extModifiers=⌥,clickCount=1] on MouseWindow$1[,0,136,200x200,layout=java.awt.FlowLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777225,maximumSize=,minimumSize=,preferredSize=]`
  * modifiers: 12 (`⌥+⌘+Button2+Button3`)
  * modifiersEx: 512 (`⌥`)

Thus, as I assumed, modifiersEx don't contain `BUTTON3_DOWN_MASK`, so we shouldn't restrict the patch by the `jmodifiers == 0` condition.